src/V3SignalStrength.cpp

```diff
@@ -0,0 +1,226 @@
+                            assignFl, strength0Varp, strength0Level,
+                            assigns[i]->rhsp()->cloneTree(false),
+                            new AstConst{assignFl, AstConst::WidthedValue(), 1, 0}));
+                        strengthBlockp->addStmtsp(getStrengthAssignmentp(
+                            assignFl, strength0Varp, strength0Level,
+                            assigns[i]->rhsp()->cloneTree(false),
+                            new AstConst{assignFl, AstConst::StringToParse(), "1'x"}));
+                    }
+                    if (strength1Level != 0) {
+                        strengthBlockp->addStmtsp(getStrengthAssignmentp(
+                            assignFl, strength1Varp, strength1Level,
+                            assigns[i]->rhsp()->cloneTree(false),
+                            new AstConst{assignFl, AstConst::WidthedValue(), 1, 1}));
+                        strengthBlockp->addStmtsp(getStrengthAssignmentp(
+                            assignFl, strength1Varp, strength1Level,
+                            assigns[i]->rhsp()->cloneTree(false),
+                            new AstConst{assignFl, AstConst::StringToParse(), "1'x"}));
+                    }
+
+                    assigns[i]->unlinkFrBack()->deleteTree();
+                }
+                nodep->addStmtp(
+                    new AstAlways{declVarFl, VAlwaysKwd::ALWAYS, nullptr, strengthBlockp});
+
+                AstVarRef* varRefp = new AstVarRef{declVarFl, varp, VAccess::WRITE};
+                nodep->addStmtp(new AstAssignW{
+                    declVarFl, varRefp,
+                    new AstCond{
+                        declVarFl,
+                        new AstGt{declVarFl,
+                                  new AstVarRef{declVarFl, strength0Varp, VAccess::READ},
+                                  new AstVarRef{declVarFl, strength1Varp, VAccess::READ}},
+                        new AstConst{declVarFl, AstConst::StringToParse(), "'0"},
+                        new AstCond{
+                            declVarFl,
+                            new AstEq{declVarFl,
+                                      new AstVarRef{declVarFl, strength0Varp, VAccess::READ},
+                                      new AstVarRef{declVarFl, strength1Varp, VAccess::READ}},
+                            new AstCond{
+                                declVarFl,
+                                new AstEq{
+                                    declVarFl,
+                                    new AstVarRef{declVarFl, strength0Varp, VAccess::READ},
+                                    new AstConst{declVarFl, AstConst::StringToParse(), "'0"}},
+                                new AstConst{declVarFl, AstConst::StringToParse(), "'z"},
+                                new AstConst{declVarFl, AstConst::StringToParse(), "'x"}},
+                            new AstConst{declVarFl, AstConst::StringToParse(), "'1"}}}});
```

Comment:
If you keep this, split it into methods so it's easier to read.

Reply:
Done